test/src/org/labkey/test/tests/idri/FormulationsTest.java

```diff
@@ -312,30 +316,21 @@ protected void defineParticleSizeAssay()
 
         log("Defining Particle Size Assay");
         clickAndWait(Locator.linkWithText("Manage Assays"));
-        clickButton("New Assay Design");
+        ReactAssayDesignerPage assayDesignerPage = _assayHelper.createAssayDesign("Particle Size", PS_ASSAY)
+            .setDescription(PS_ASSAY_DESC);
 
-        assertTextPresent("Particle Size Data");
-        checkCheckbox(Locator.radioButtonByNameAndValue("providerName", "Particle Size"));
-        clickButton("Next");
+        assayDesignerPage.goToBatchFields();
+        assertElementPresent(Locator.tagWithClass("div", "domain-form-no-field-panel"));
 
-        waitForElement(Locator.xpath("//input[@id='AssayDesignerName']"), WAIT_FOR_JAVASCRIPT);
-        setFormElement(Locator.xpath("//input[@id='AssayDesignerName']"), PS_ASSAY);
-        setFormElement(Locator.xpath("//textarea[@id='AssayDesignerDescription']"), PS_ASSAY_DESC);
-        fireEvent(Locator.xpath("//input[@id='AssayDesignerName']"), SeleniumEvent.blur);
+        DomainFormPanel runPanel = assayDesignerPage.goToRunFields();
+        assertThat("Missing expected run property", runPanel.getField("IDRIBatchNumber"), CoreMatchers.notNullValue());
 
+        DomainFormPanel resultPanel = assayDesignerPage.goToFieldProperties("Result Properties");
+        assertThat("Missing expected result property", resultPanel.getField("MeasuringTemperature"), CoreMatchers.notNullValue());
+        assertThat("Missing expected result property", resultPanel.getField("meanCountRate"), CoreMatchers.notNullValue());
+        assertThat("Missing expected result property", resultPanel.getField("AnalysisTool"), CoreMatchers.notNullValue());
```

Comment:
Use `DomainFormPanel.fieldNames()` to verify expected fields in bulk instead of one-by-one null checks:
```
DomainFormPanel batchPanel = assayDesignerPage.goToBatchFields();
assertEquals("Batch fields", Collections.emptyList(), batchPanel.fieldNames());

DomainFormPanel runPanel = assayDesignerPage.goToRunFields();
assertEquals("Run fields", Arrays.asList("IDRIBatchNumber"), batchPanel.fieldNames());

DomainFormPanel resultPanel = assayDesignerPage.goToResultFields();
assertEquals("Result fields", Arrays.asList("MeasuringTemperature", "meanCountRate", "AnalysisTool"), resultPanel.fieldNames());
```
If there might be extra fields that we don't care about, you can use something like:
```
assertThat("Result fields", resultPanel.fieldNames(), CoreMatchers.hasItems("MeasuringTemperature", "meanCountRate", "AnalysisTool"));
```

Reply:
Done in new fb.